src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +283,103 @@ public void addSeparator() {
 		public void addEndButton(JButton button) {
 			messagePanel.addOptions(button, HttpPanel.OptionsLocation.END);
 		}
+
+		private JButton getBtnReopen(WebSocketPanelSender sender) {
+			if (btnReopen == null) {
+				btnReopen = new JButton();
+				btnReopen.setIcon(REOPEN_ICON);
+				btnReopen.setEnabled(true);
+				btnReopen.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_hint"));
+				btnReopen.addActionListener(e -> {
+					if(wsUiHelper.isAlwaysGenerate()){
+						newSecWebSocketKey = WebSocketUtils.generateSecWebSocketKey();
+					}else {
+						newSecWebSocketKey = (wsUiHelper.isGenerateKeyPressed() ? wsUiHelper.getWebSocketKey() : null);
+					}
+					PersistentConnectionListener persistentConnectionListener = new PersistentConnectionListener() {
+						@Override
+						public boolean onHandshakeResponse(HttpMessage httpMessage, Socket socket, ZapGetMethod zapGetMethod) {
+
+							return false;
+						}
+
+						@Override
+						public int getArrangeableListenerOrder() {
+							return 0;
+						}
+					};
+					if(!sender.reOpenChannel(wsUiHelper.getSelectedChannelId(), newSecWebSocketKey, wsUiHelper.isRedirection(), wsUiHelper.isTrackingSession())){
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.unable_reopen"));
+					}
+				});
+			}
+			return btnReopen;
+		}
+
+		private JButton getBtnReopenEdit(WebSocketPanelSender sender){
+			if(btnReopenEdit == null) {
+				btnReopenEdit = new JButton();
+				btnReopenEdit.setIcon(REOPEN_EDIT_ICON);
+				btnReopenEdit.setEnabled(true);
+				btnReopenEdit.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_edit_hint"));
+				btnReopenEdit.addActionListener(e -> {
+					try {
+						wsUiHelper.setSecWebsocketKeyField(sender.getSecWebKeyFrom(wsUiHelper.getSelectedChannelId()));
+						JOptionPane.showMessageDialog(this, getReopenPanel(),
+								Constant.messages.getString("websocket.manual_send.adv_dialog.title"),
+								JOptionPane.PLAIN_MESSAGE);
+					} catch (IllegalArgumentException e1) {
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.invalid_channel"));
+					} catch (DatabaseException e1) {
+						LOGGER.info(e1.getStackTrace().toString());
```

Comment:
We should inform the user with a dialogue in this case too (this and following catch could be merged).